sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -203,6 +203,70 @@ case class GpuTimeSub(start: Expression,
   }
 }
 
+case class GpuDateAddInterval(start: Expression,
+    interval: Expression,
+    timeZoneId: Option[String] = None)
+    extends GpuTimeMath(start, interval, timeZoneId) {
+
+  override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression = {
+    copy(timeZoneId = Option(timeZoneId))
+  }
+
+  override def intervalMath(us_s: Scalar, us: ColumnVector): ColumnVector = {
+    us.add(us_s)
+  }
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(DateType, CalendarIntervalType)
+
+  override def dataType: DataType = DateType
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+    var lhs: Any = null
+    var rhs: Any = null
+    try {
+      lhs = left.columnarEval(batch)
+      rhs = right.columnarEval(batch)
+
+      (lhs, rhs) match {
+        case (l: GpuColumnVector, intvl: CalendarInterval) =>
+          if (intvl.months != 0) {
+            throw new UnsupportedOperationException("Months aren't supported at the moment")
+          }
+          val microSecToDays = if (intvl.microseconds < 0) {
+            // This is to calculate when subtraction is performed. Need to take into account the
+            // interval( which are less than days). Convert it into days which needs to be
+            // subtracted along with intvl.days(if provided).
+            (intvl.microseconds.abs / (24 * 60 * 60 * 1000 * 1000.0)).ceil.toInt * -1
```

Comment:
it would be more readable to convert the numerator `.toDouble` instead of sneaking in a double as the last factor in the denominator. 

Can we compute/define the number of microseconds in a day `24 * 60 * 60 * 1000 * 1000` only once,  and reuse the constant. Instead of using our own let us use [TimeUnit.DAYS.toMicros(1)](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/TimeUnit.html#toMicros-long-)